src/api/projectApi.ts

```diff
@@ -1,31 +1,16 @@
-import Axios, { CancelToken } from 'axios';
-
-import i18n from '../translations/i18n';
-import { StatusCode } from '../utils/constants';
-import { getIdToken } from './userApi';
+import { CancelToken } from 'axios';
+import { CristinProject } from '../types/project.types';
+import { authenticatedApiRequest } from './apiRequest';
 
 export enum ProjectsApiPaths {
   PROJECT = '/project',
 }
 
 export const searchProjectsByTitle = async (query: string, cancelToken?: CancelToken) => {
-  const titleQuery = `title=${encodeURIComponent(query)}`;
-  try {
-    const idToken = await getIdToken();
-    const response = await Axios.get(`${ProjectsApiPaths.PROJECT}?${titleQuery}`, {
-      headers: {
-        Authorization: `Bearer ${idToken}`,
-      },
-      cancelToken,
-    });
-    if (response.status === StatusCode.OK) {
-      return response.data;
-    } else {
-      return { error: i18n.t('feedback:error.get_project') };
-    }
-  } catch (error) {
-    if (!Axios.isCancel(error)) {
-      return { error: i18n.t('feedback:error.get_project') };
-    }
-  }
+  const url = `${ProjectsApiPaths.PROJECT}?title=${encodeURIComponent(query)}`;
```

Comment:
hvorfor ikke bare slenge encodeURIComponent rundt hele url - så er man safe om det kommer inn flere params

Reply:
det blir ikke det samme: 
![bilde](https://user-images.githubusercontent.com/6313468/102338990-d8351800-3f94-11eb-99ff-406aac4d54cc.png)

men nå ble jeg veldig usikker på hva Axios gjør selv behind the scenes... er det ikke noe som egentlig burde gjøres der så man slipper ekstra styr med nye params? må undersøke dette litt mer

Reply:
jeg mente encodeURI, ja
Jeg venter i spenning :-)

Reply:
Ser ut som vi har slurvet litt med Axios-måten å gjøre ting på, tror dette er bedre
```
export const searchProjectsByTitle = async (query: string, cancelToken?: CancelToken) => {
  return authenticatedApiRequest<CristinProject[]>({
    url: ProjectsApiPaths.PROJECT,
    params: {
      title: query,
    },
    cancelToken,
  });
};
```

Angående kommentaren din på den andre PRen om å encode her: ` url: '${PublicationsApiPaths.PUBLICATION}/${identifier}/publish'`, så ser jeg i network-tab at mellomrom blir `%20` både med og usen `encodeURIComponent`, så det ser faktisk ut som at Axios fikser alt selv... 🤔 men finner ikke noe dokumentasjon på at axios fikser det, men finner heller ikke noen plass det står at man bør gjøre det selv. sååå heller mot å droppe all encoding (til det evt blir et problem)

Reply:
all encoding i api-kall verdt å merke. vi har URI-er som id i URL-en noen steder, og der må vi bruke `encodeURI` selv